### DISCUSSION TITLE
how to do i create a table with multiple column with check boxes

### DISCUSSION THREAD

**Discussion by TechMARK-1978** [primefaces/community#212] [PrimeNG]
i want to give rights to user for that i have a table with the name of 
1-Menu Name
2-AddRight
3-EditRight
4-DeleteRight 
5-ViewRight 
for that i code the following

<p-table [value]="MenuList" [tableStyle]="{'min-width': '50rem'}">
                        <ng-template pTemplate="header" let-columns>
                            <tr>
                                <th>Name </th>
                                <th>Add</th>
                                <th>Edit</th>
                                <th>Delete</th>
                                <th>View</th>
                                <th>MenuView</th>
                            </tr>
                        </ng-template>
                        <ng-template pTemplate="body" let-MenuList let-columns="columns">
                            <tr>
                                <td>{{MenuList.nam}}</td>
                                <td><p-tableCheckbox value={{MenuList.AddRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.EditRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.DeleteRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.ViewRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.MenuView}}></p-tableCheckbox></td>
                            </tr>
                        </ng-template>
                    </p-table>

after that when i click on any columns check box all automatically checked plz guide me in this regard i am using reactive form
Thanks in Advance

**Comment by TechMARK-1978** [comment]
`<p-table [value]="MenuList" [tableStyle]="{'min-width': '50rem'}">
                        <ng-template pTemplate="header" let-columns>
                            <tr>
                                <th>Name </th>
                                <th>Add</th>
                                <th>Edit</th>
                                <th>Delete</th>
                                <th>View</th>
                                <th>MenuView</th>
                            </tr>
                        </ng-template>
                        <ng-template pTemplate="body" let-MenuList let-columns="columns">
                            <tr>
                                <td>{{MenuList.nam}}</td>
                                <td><p-tableCheckbox value={{MenuList.AddRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.EditRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.DeleteRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.ViewRight}}></p-tableCheckbox></td>
                                <td><p-tableCheckbox value={{MenuList.MenuView}}></p-tableCheckbox></td>
                            </tr>
                        </ng-template>
                    </p-table>`

**Comment by cetincakiroglu** [comment]
Hi,

Please create a Stackblitz example so we can play with your code and help you, sharing code like this doesn't help to understand the point or identify the issues.
